tb_plugin/torch_tb_profiler/io/cache.py

```diff
@@ -37,17 +40,31 @@ def __setstate__(self, state):
         self.__dict__.update(state)
 
     def read(self, filename):
-        local_file = self._cache_dict.get(filename)
+        local_file = self._gpu_metrics_cache_dict.get(filename)
         if local_file is None:
-            local_file = download_file(filename)
-            # skip the cache for local files
-            if local_file != filename:
-                with self._lock:
-                    self._cache_dict[filename] = local_file
+            logger.debug("_gpu_metrics_cache_dict.get({}) is None".format(filename))
+            local_file = self._cache_dict.get(filename)
+            if local_file is None:
+                local_file = download_file(filename)
+                # skip the cache for local files
+                if local_file != filename:
+                    with self._lock:
+                        self._cache_dict[filename] = local_file
 
         logger.debug("reading local cache %s for file %s" % (local_file, filename))
         with File(local_file, 'rb') as f:
-            return f.read()
+            return f.read(), filename in self._gpu_metrics_cache_dict
+
+    def write_gpu_metrics(self, raw_data, filename):
+        fp = tempfile.NamedTemporaryFile('w+b', suffix='.json.gz', delete=False)
+        fp.close()
```

Comment:
basically, the cache class has no idea about the gpu metrics. Instead, these logic should be moved out of this cache class.

It should simply has some logic like: add_local_cache(source_file, dest_file), which is called in the plugin.py

Reply:
actually, the trace view route doesn't need the cache class at all since a simply self._trace_file in plugin.py is enough for the functionality.

Reply:
Sounds good. Thanks. Done.